src/NamedTupleUtilities.jl

```diff
@@ -0,0 +1,136 @@
+"""
+    select(a::NamedTuple, v::Val{n})
+Select a field `n` from `a` if it is in `a`.
+```jldoctest
+julia> select((a=1,b=2,c=3),Val(:a))
+(a = 1,)
+```
+"""
+@generated function select(a::NamedTuple{an}, ::Val{bn}) where {an, bn}
+    names = ((i for i in an if i == bn)...,)
+    types = Tuple{(fieldtype(a, n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :(NamedTuple{$names,$types}(($(vals...),)))
+end
+
+"""
+    remove(a::NamedTuple, v::Val{n})
+Remove a field `n` from the `a` if it is in `a`.
+```jldoctest
+julia> remove((a=1,b=2,c=3),Val(:c))
+(a = 1, b = 2)
+```
+"""
+@generated function remove(a::NamedTuple{an}, ::Val{bn}) where {an, bn}
+    names = ((i for i in an if i != bn)...,)
+    types = Tuple{(fieldtype(a ,n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :(NamedTuple{$names,$types}(($(vals...),)))
+end
+
+"""
+    range(a::NamedTuple, b::Val{n}, c::Val{n})
+Return a NamedTuple which retains the fields from `b` to `c` in `a`. 
+If `b` is not in `a`, then it will return empty NamedTuple. 
+If `c` is not in `a`, then it will return everything started with `b`.
+```jldoctest
+julia> range((a=1,b=2,c=3),Val(:a),Val(:b))
+(a = 1, b = 2)
+```
+"""
+@generated function Base.range(a::NamedTuple{an}, ::Val{bn}, ::Val{cn}) where {an, bn, cn}
```

Comment:
I think we should probably not extend ``Base.range`` here, but instead just define a new ``range`` function... Otherwise we might run into conflicts down the road. So just remove the ``Base.`` part here.